src/Core/Framework/Adapter/Translation/Translator.php

```diff
@@ -343,10 +343,10 @@ private function getCustomizedCatalogue(MessageCatalogueInterface $catalog, ?str
     private function loadSnippets(MessageCatalogueInterface $catalog, string $snippetSetId, ?string $fallbackLocale): array
     {
         $this->resolveSalesChannelId();
+        $effectiveLocale = $fallbackLocale ?? $catalog->getLocale();
+        $key = \sprintf('translation.catalog.%s.%s', $this->salesChannelId ?: 'DEFAULT', $snippetSetId . '-' . $effectiveLocale);
 
-        $key = \sprintf('translation.catalog.%s.%s', $this->salesChannelId ?: 'DEFAULT', $snippetSetId);
-
-        return $this->cache->get($key, function (ItemInterface $item) use ($catalog, $snippetSetId, $fallbackLocale) {
+        return $this->cache->get($key, function (ItemInterface $item) use ($catalog, $snippetSetId, $effectiveLocale) {
             if (Feature::isActive('cache_rework')) {
```

Comment:
the POC should be rebased, the cache_rework is already done

Reply:
done